### PR TITLE
Fixes for running EMEP report

### PR DESCRIPTION
**Pull request by ejgal**
Some features in ReadMscwCtm got lost in a merge at some point.
This PR adds some tests and fixes for the missing features.

### CODE REVIEW

File: pyaerocom/io/read_mscw_ctm.py [hunk 1]
```diff
@@ -11,6 +11,8 @@
 import sys
 import os
 import glob
+
+from xarray.coding.times import infer_calendar_name
```
Comment (ejgal) [inline pyaerocom/io/read_mscw_ctm.py:15]:
```suggestion
```

**Comment by ejgal** [comment]
@jgliss, can you have a look at this please?
This was needed to get the rerun of EMEP report working.

File: pyaerocom/io/read_mscw_ctm.py [hunk 2]
```diff
@@ -255,6 +259,16 @@ def read_var(self, var_name, ts_type=None, **kwargs):
         GriddedData
         """
 
+        if isinstance(ts_type, dict):
```
Comment (jgliss) [inline pyaerocom/io/read_mscw_ctm.py:262]: In which case is ts_type a dictionary here? Can this happen in the higlevel tools? I.e. in `Colocator`?

File: pyaerocom/io/read_mscw_ctm.py [hunk 3]
```diff
@@ -302,6 +318,8 @@ def read_var(self, var_name, ts_type=None, **kwargs):
             data.time.attrs['standard_name'] = 'time'
             data.attrs['units'] = self.preprocess_units(data.units, EMEP_prefix)
             cube = data.to_iris()
+            if ts_type == 'hourly':
+                cube.coords()[0].convert_units('hours since 1900-01-01')
```
Comment (jgliss) [inline pyaerocom/io/read_mscw_ctm.py:322]: Is it certain that first index is **always** the time dimension?

Comment (jgliss) [reply pyaerocom/io/read_mscw_ctm.py:322]: Maybe it is better to try `cube.coord('time')` to make sure you are acting on the time dimension.

File: pyaerocom/io/read_mscw_ctm.py [hunk 4]
```diff
@@ -324,6 +342,8 @@ def preprocess_units(units, prefix=None):
         new_unit = units
         if units == '' and prefix == 'AOD': #
             new_unit = '1'
+        elif units == '' and prefix == 'AbsCoef':
+            new_unit = '1/m'
         elif units == 'mgS/m2' or units == 'mgN/m2':
             raise NotImplementedError('Species specific units are not implemented.')
```
Comment (jgliss) [inline pyaerocom/io/read_mscw_ctm.py:348]: I think this is "sort of" implemented now so it may be okay to remove this "elif" for N and S units.

File: pyaerocom/io/test/test_read_mscw_ctm.py [hunk 5]
```diff
@@ -37,6 +37,13 @@ def test_read_emep_read_var(path_emep):
         r = ReadMscwCtm(data_dir=data_dir)
         r.read_var('vmro3', 'daily')
 
+@testdata_unavail
+def test_ts_type_dict(path_emep):
+    data_dir = path_emep['data_dir']
+    r = ReadMscwCtm(data_dir=data_dir)
+    data = r.read_var('vmro3', ts_type={'vmro3':'daily'})
+    assert data.ts_type == 'daily'
```
Comment (jgliss) [inline pyaerocom/io/test/test_read_mscw_ctm.py:45]: I still do not understand why one would pass ts_type as dict in `read_var` and if this happens somewhere in the higher level code, it should probably be fixed there (if possible). @ejgal can you check that?

**Review by jgliss — COMMENTED** [review]
All good, but I have some minor comments, maybe you could check them first, thanks!

File: pyaerocom/io/test/test_read_mscw_ctm.py (hunk 5 above)
Comment (ejgal) [reply pyaerocom/io/test/test_read_mscw_ctm.py:45]: It's from Colocator:
```
model_ts_type_read : :obj:`str` or :obj:`dict`, optional
        may be specified to explicitly define the reading frequency of the
        model data. Not to be confused with :attr:`ts_type`, which specifies
        the frequency used for colocation. Can be specified variable specific
        by providing a dictionary.
```
This can be passed all the way to the reader in some cases.
In ReadGridded._eval_vert_which there is also a check for ts_type being a dict, so there might be a usecase for this.

We could make sure to extract the ts_type for the variable with:
```
if isinstance(kwargs['ts_type'], dict):
    kwargs['ts_type'] = kwargs['ts_type'][var_name]
```
right before calling read_var.

Doing this does not break any of the tests at least.

File: pyaerocom/io/read_mscw_ctm.py (hunk 3 above)
Comment (ejgal) [reply pyaerocom/io/read_mscw_ctm.py:322]:
```suggestion
                cube.coords('time').convert_units('hours since 1900-01-01')
```

Comment (ejgal) [reply pyaerocom/io/read_mscw_ctm.py:322]: Good point. Not sure how much the EMEP files can vary, this seems like a better solution.

File: pyaerocom/io/read_mscw_ctm.py (hunk 4 above)
Comment (ejgal) [inline pyaerocom/io/read_mscw_ctm.py:348]:
```suggestion
```

File: pyaerocom/io/test/test_read_mscw_ctm.py (hunk 5 above)
Comment (ejgal) [reply pyaerocom/io/test/test_read_mscw_ctm.py:45]: Should I try to fix this as proposed, or do we leave it for now @jgliss?

File: pyaerocom/io/read_mscw_ctm.py (hunk 4 above)
Comment (ejgal) [reply pyaerocom/io/read_mscw_ctm.py]: I tested it now, and I think the "sort of" implementation we did is not included in the release_0_10_0 branch.

File: pyaerocom/io/test/test_read_mscw_ctm.py (hunk 5 above)
Comment (jgliss) [reply pyaerocom/io/test/test_read_mscw_ctm.py:45]: Feel free to update as proposed, but please make an issue for it, so I can check at some point later what was the idea behind it :) ? It looks like this is for some edge cases and I am not sure anymore. But it seems it is also placed wrongly in if it is evaluated in `ReadGridded._eval_vert_which` ...

File: pyaerocom/io/read_mscw_ctm.py (hunk 4 above)
Comment (ejgal) [reply pyaerocom/io/read_mscw_ctm.py:339]: I added this back in for now.

**Comment by ejgal** [comment]
This is updated now, can you have a brief look @jgliss?

File: pyaerocom/io/test/test_read_mscw_ctm.py (hunk 5 above)
Comment (ejgal) [reply pyaerocom/io/test/test_read_mscw_ctm.py]: Done: #276

File: pyaerocom/io/read_mscw_ctm.py [hunk 6]
```diff
@@ -302,6 +308,8 @@ def read_var(self, var_name, ts_type=None, **kwargs):
             data.time.attrs['standard_name'] = 'time'
             data.attrs['units'] = self.preprocess_units(data.units, EMEP_prefix)
             cube = data.to_iris()
+            if ts_type == 'hourly':
+                cube.coords('time').convert_units('hours since 1900-01-01')
```
Comment (jgliss) [inline pyaerocom/io/read_mscw_ctm.py:312]: I don't think `cube.coords('time')` works (cube.coords method returns a list I think), I think the syntax is `cube.coord('time')`. Please check that. Thanks.
```suggestion
                cube.coord('time').convert_units('hours since 1900-01-01')
```

File: pyaerocom/io/read_mscw_ctm.py [hunk 7]
```diff
@@ -324,9 +332,11 @@ def preprocess_units(units, prefix=None):
         new_unit = units
         if units == '' and prefix == 'AOD': #
             new_unit = '1'
+        elif units == '' and prefix == 'AbsCoef':
+            new_unit = '1/m'
```
Comment (jgliss) [inline pyaerocom/io/read_mscw_ctm.py:336]:
```suggestion
            new_unit = 'm-1'
```
just for consistency, I think we should avoid /unit and rather use unit-1 nomenclature.

**Review by jgliss — CHANGES_REQUESTED** [review]
Can you please check the minor issues? Cheers!

**Review by jgliss — APPROVED** [review]
Looks good to me =) Takk!

**Comment by ejgal** [comment]
Thanks! I tested .coord interactively, but of course ended up writing coords ;)

**Comment by jgliss** [comment]
> Thanks! I tested .coord interactively, but of course ended up writing coords ;)

That's why we should always review each other's changes =)
